tests/handlers/test_agent.py

```diff
@@ -30,3 +34,42 @@ def test_post_agent_unauthenticated(test_client, db_session):
 
     assert response.status_code == 403
     assert response.json()['detail'] == "Permission denied."
+
+
+def test_list_agent_empty_list(db_session, test_client, session_cookie):
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.json() == []
+
+
+def test_list_agent_success(db_session, test_client, session_cookie):
+    agent = Agent(name='test', oauth_client=OAuthClient(type='confidential'))
+    db_session.add(agent)
+    db_session.commit()
+
+    response = test_client.get('/agent', cookies=session_cookie)
+
+    assert response.status_code == 200
+    response_body = response.json()
+
+    assert len(response_body) == 1
+
+    assert response_body[0]['name'] == 'test'
+    # assert response_body[0]['connection_type'] == 'd'
+    oauth_client = response_body[0]['oauth_client']
+    assert oauth_client['agent_id'] is not None
+    assert oauth_client['secret'] is not None
+    assert oauth_client['type'] == 'confidential'
+
+    agent = Agent(name='test2', oauth_client=OAuthClient(type='confidential'))
+    db_session.add(agent)
+    db_session.commit()
+
+    response = test_client.get('/agent', cookies=session_cookie)
```

Comment:
I'm kinda confused by this call. It goes unasserted so you might as well not make it. I'd just add two agents right away and assert that the response size is 2 and go into detail on one of the objects.